elementary/monitor/dbt_project/macros/get_test_results.sql

```diff
@@ -57,8 +58,8 @@
             {% if not disable_passed_test_metrics %}
                 {% do elementary_tests_allowlist_status.append('pass') %}
             {% endif %}
-            {% set test_rows_sample = elementary_internal.get_test_rows_sample(test_result_rows_agate.get(test.id), test_type, metrics_sample_limit) %}
             {%- if (test_type == 'dbt_test' and status in ['fail', 'warn']) or (test_type != 'dbt_test' and status in elementary_tests_allowlist_status) -%}
+                {% set test_rows_sample = elementary_internal.get_test_rows_sample(test.result_rows, test_result_rows_agate.get(test.id), test_type, test.test_results_query, metrics_sample_limit) %}
```

Comment:
What is the reason for moving it to be inside the "if"?
(and in general not sure I understand the purpose of this "if" - seems like it only does logic specific to dimension anomalies?)

Reply:
Error tests are not supposed to be sampled.
The reason it worked so far is because [error tests](https://github.com/elementary-data/dbt-data-reliability/commit/8f312b1c369dd3608a3af9f7273ab0f36438590b#diff-7e7a0c1e6c6911e382b5c70e89456fce2aa74e4da96f1cd0295733468df8d563R29) default to [empty list `result_rows`](https://github.com/elementary-data/dbt-data-reliability/commit/3a8aca03fc126a6b13a7fc0a5bbf98ea812de3a7).

However, now a user that uses `test_result_rows`, this is what will happen on `get_test_rows_sample`:
1. `elementary_test_results.result_rows` will be `null`, so it'll progress.
2. `elementary_test_results.test_result_rows` will not contain any samples because the test materialization failed.
3. It'll attempt to run the query and crash because it's broken.
Point being, error tests should not be sampled.